src/planner/dart/ConfigurationToConfiguration_to_ConfigurationToTSR.cpp

```diff
@@ -84,24 +87,57 @@ ConfigurationToConfiguration_to_ConfigurationToTSR::plan(
   auto saver = MetaSkeletonStateSaver(mMetaSkeleton);
   DART_UNUSED(saver);
 
+  auto robot = mMetaSkeleton->getBodyNode(0)->getSkeleton();
+
+  std::vector<MetaSkeletonStateSpace::ScopedState> configurations;
+
+  // Use a ranker
+  ConstConfigurationRankerPtr configurationRanker(mConfigurationRanker);
```

Comment:
Why do we need to create another pointer here? Can we not do:
```
if (!mConfigurationRanker)
{
    mConfigurationRanker = NominalConfigurationRanker;
}
```
In `robot/utils/planToTSR`, the ranker has been taken with a `const` pointer so I guess that's fine there. Alternatively, we can leave this like this and change the data member to `const` since we are initializing it in the constructor anyway.

Reply:
Summarizing my slack discussion with @aditya-vk , I think we should keep it as is, for the following reasons:
1) if the planner is created with `mConfigurationRanker = nullptr`, the expectation is that a NominalConfigurationRanker would be created when this method is called and be initialized with the `startState`. So if we change the member variable as you suggested, the planner now has been mutated and it cannot be reused in the same way. Next time it is called, it'd use this ranker. 
2) To fix, the ideal solution would be the second way you suggested, to have it as `const mConfigurationRanker` so that it never gets changed. Unfortunately, to my best understanding, because this class _becomes_ a TSRPlanner via the templated adapter, it does not have access to `mConfigurationRanker` during the initialization, i.e. I can't do `: mConfigurationRanker(std::move(ranker)) { ... } `. It only has access to the constructor parameters of the PlannerAdapter, which is the `stateSpace` and `metaSkeleton`. See [stackoverflow](https://stackoverflow.com/questions/2290733/initialize-parents-protected-members-with-initialization-list-c) .

Hence, in order to keep the member variable as-is (as if it's a const) _and_ initialize it in constructor if it's passed in, I think we need to keep it the way I have it now. 

@aditya-vk @brianhou  please confirm!

Reply:
That's right, there's no way to have it in the initialization list. Hmm.
I think what we have now is good then. Since we don't give an option to change the ranker at all unless a new planner is created, I think it should be fine :+1: